src/penn_chime/validators/validators.py

```diff
@@ -0,0 +1,69 @@
+"""design pattern via https://youtu.be/S_ipdVNSFlo?t=2153"""
+
+from typing import Optional
+from datetime import date, datetime
+
+from .base import Validator
+
+
+class Bounded(Validator):
+    """A bounded number."""
+    def __init__(
+            self,
+            lower_bound: Optional[float] = None,
+            upper_bound: Optional[float] = None) -> None:
+        assert lower_bound is not None or upper_bound is not None, "Do not use this object to create an unbounded validator."
+        self.lower_bound = lower_bound
+        self.upper_bound = upper_bound
+        self.message = {
+            (lower_bound, upper_bound): f"in ({self.lower_bound}, {self.upper_bound})",
+            (None, upper_bound): f"less than {self.upper_bound}",
+            (lower_bound, None): f"greater than {self.lower_bound}",
```

Comment:
If either one of them is `None`, the first case still overlaps with one of the latter two.

Reply:
i'm not an expert in python dict initialization, but wouldn't the second key,val pair in the "list" take precedence? 

do you think i should do it in 
```
self.message = dict()
self.message[(lower_bound, upper_bound)] = "message1" 
self.message[(None, upper_bound)] = "message1"
...```
style instead?

Reply:
```
>>> x = {1: 'a', 2: 'b', 1: 'c'}
>>> x[1]
'c'
```
as i expected

its still possible i'm _really_ turned around here and not reading my own code right.

Reply:
Ok, it works as intended, but reading it relies on knowing that quirk of Python dictionary initialization. I think it's more important to get this merged than to argue on this point of style, so I'll Approve.